nym-vpn-core/crates/nym-vpn-network-config/src/refresh.rs

```diff
@@ -134,12 +140,31 @@ impl FileRefresher {
 }
 
 // Ideally we only refresh the discovery file when the tunnel is up
-pub fn start_background_file_refresh(
+pub async fn start_background_file_refresh(
     config_path: PathBuf,
     network: Network,
-    discovery_refresher_tx: tokio::sync::mpsc::Sender<Result<Network>>,
+    discovery_refresher_tx: tokio::sync::mpsc::Sender<FileRefresherEvent>,
     cancel_token: CancellationToken,
-) -> JoinHandle<()> {
-    let refresher = FileRefresher::new(config_path, discovery_refresher_tx, cancel_token);
-    tokio::spawn(refresher.run(network))
+) -> Result<JoinHandle<()>> {
+    let (client, resolver_overrides) = Discovery::create_client().await?;
```

Comment:
I think we should reuse the `resolver_overrides`  by passing them in the  `start_background_file_refresh` call and don't mess with the firewall again